hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/test/TestSourcesTest.java

```diff
@@ -74,10 +74,24 @@ public void test_itemStream_withWindowing() throws Throwable {
          .withNativeTimestamps(0)
          .window(WindowDefinition.tumbling(1000))
          .aggregate(AggregateOperations.counting())
-         .apply(assertCollectedEventually(10, windowResults -> {
-             // find any window that has 10 items, some may be incomplete due to hiccups
-             boolean matched = windowResults.stream().anyMatch(r -> r.result() == itemsPerSecond);
-             assertTrue("Did not find any window with 10 items: " + windowResults, matched);
+         .apply(assertCollectedEventually(60, windowResults -> {
+             //look at last 5 windows at most
+             int windowsToConsider = Math.min(5, windowResults.size());
+
+             //count the total no. of items emitted in those windows
+             int totalItems = windowResults.stream()
+                     .skip(windowResults.size() - windowsToConsider)
```

Comment:
this will throw `IllegalArgumentException` because you can get a negative number here

Reply:
I don't see how I could get a negative number there, until `windowsToConsider = Math.min(5, windowResults.size())`.

Reply:
yes, you are right, my bad